### PR TITLE
NDVI files for review

### PR DESCRIPTION
**Pull request by lrwives**
# Pull Request Template

## CI Checks
- [ ] The notebook runs from start to finish on all operating systems:
   - [ ] Mac
   - [ ] Windows
   - [ ] Linux 
### Reproducibility
- [ ] Are the data downloaded in the code
- [ ] Are paths created to ensure they work on all operating systems (using  os.path.join)
- [ ] Are comments used to clarify the contents of the code that can’t be clarified using expressive variable and function names alone? (not too many comments - just enough)
- [ ] Does the notebook run from start to finish?

### Functions
- [ ] Do functions follow PEP 8 format conventions?
- [ ] Are function docstrings clear (all  inputs and outputs clearly  described and defined)
- [ ] Are function names expressive (the name describes what the function does)?
- [ ] Are functions easy to understand and read?
- [ ] How many tasks does each function do? (ideally a function does one thing well).
### Package imports
- [ ] Are standard modules (those included with the base  python install) vs. third party (related but externally developed tools) import groups correct with appropriate spacing in between each group?
- [ ] Are variable names throughout the code, expressive?
- [ ] Suggest changes if not, highlight what was done well
- [ ] Is the code overall easy to understand and read? Are there things that would make it more clear / cleaner?

### DRY Code
- [ ] Are segments of code repeated in the file or is the code DRY?
- [ ] Are loops used to optimize DRY code?
- [ ] Are functions used to optimize DRY code?
- [ ] Are there any areas that could be potentially improved (you can suggest improvements OR you can just highlight parts of the code where you suspect it could  be cleaner / more efficient.

### Novel Approaches to Problem solving
- [ ] Highlight any novel approaches to completing the assignment.

### CODE REVIEW

**Comment by gnarledbranches** [comment]
> # Pull Request Template
> ## Review Checklist
> ## CI Checks
> * [ ]  The notebook runs from start to finish on all operating systems: _(Not sure if this is a Github check issue, or an issue with student's notebooks in actuality, but they seem to not run on all systems.)_
>   
>   * [ ]  Mac
>   * [ ]  Windows
>   * [ ]  Linux
> 
> ### Reproducibility
> * [x]  Are the data downloaded in the code          _yes, all downloads work_
> * [x]  Are paths created to ensure they work on all operating systems (using  os.path.join) _yes, os.path.join is used, but see CI checks comment above_
> * [x]  Are comments used to clarify the contents of the code that can’t be clarified using expressive variable and function names alone? (not too many comments - just enough) _yes, comments are clean and concise_
> * [x]  Does the notebook run from start to finish? _yes_
> 
> ### PEP 8 standards & Code Readability
> ### Functions
> * [x]  Do functions follow PEP 8 format conventions? _yes_
> * [x]  Are function docstrings clear (all  inputs and outputs clearly  described and defined) _yes_
> * [x]  Are function names expressive (the name describes what the function does)? _yes, very clear_
> * [x]  Are functions easy to understand and read? _yes_
> * [x]  How many tasks does each function do? (ideally a function does one thing well). _one clear task per function_
> 
> ### Package imports
> * [x]  Are standard modules (those included with the base  python install) vs. third party (related but externally developed tools) import groups correct with appropriate spacing in between each group?
> * [x]  Are variable names throughout the code, expressive? _yes!_
> * [x]  Suggest changes if not, highlight what was done well _n/a_
> * [x]  Is the code overall easy to understand and read? Are there things that would make it more clear / cleaner?
> 
> ### DRY Code
> * [x]  Are segments of code repeated in the file or is the code DRY? _no repeats_
> * [x]  Are loops used to optimize DRY code? _yes_
> * [x]  Are functions used to optimize DRY code? _yes_
> * [x]  Are there any areas that could be potentially improved (you can suggest improvements OR you can just highlight parts of the code where you suspect it could  be cleaner / more efficient. _no improvements to structure/format of code, but some adjustments could have been made to fix the functionality in regard to calculating NDVI correctly and removing NA values. It also looks like your pseudocode cell was missing up top where you could have broken down workflow. But your code was so clean that I imagine this wouldn't be an issue outside of an educational environment_
> 
> ### Novel Approaches to Problem solving
> * [x]  Highlight any novel approaches to completing the assignment. _great job overall! Your functions and loops were very neat, concise, and readable. I could learn from this!_

File: ives_libby_ndvi.py [hunk 1]
```diff
@@ -0,0 +1,1615 @@
...
+     "grade_id": "pseudo-code",
+     "locked": true,
+     "points": 15,
+     "schema_version": 3,
+     "solution": false,
+     "task": true
+    }
+   },
+   "source": [
+    "# Replace this cell with your pseudocode  for this workflow\n",
```
Comment (gnarledbranches) [inline ives_libby_ndvi.py:157]: @lrwives This cell was not replaced with pseudocode for your workflow.

File: ives_libby_ndvi.py [hunk 2]
```diff
@@ -0,0 +1,1615 @@
...
+    "import geopandas as gpd\n",
+    "from shapely.geometry import mapping\n",
+    "import rasterio as rio\n",
+    "import xarray as xr\n",
+    "import rioxarray as rxr\n",
+    "import earthpy as et\n",
+    "import earthpy.spatial as es\n",
+    "import earthpy.plot as ep\n",
+    "import earthpy.mask as em\n",
+    "from matplotlib.dates import DateFormatter\n",
```
Comment (gnarledbranches) [inline ives_libby_ndvi.py:273]: Great! Imports listed at top and organized.

File: ives_libby_ndvi.py [hunk 3]
```diff
@@ -0,0 +1,1615 @@
...
+     "task": false
+    }
+   },
+   "outputs": [],
+   "source": [
+    "def calculate_mean_ndvi_clouds(folder_path):\n",
+    "    '''\n",
+    "\n",
+    "    Parameters\n",
+    "    ----------\n",
```
Comment (gnarledbranches) [inline ives_libby_ndvi.py:471]: Missing a brief function description string here , before parameters, summarizing each of the steps commented out below. (As you did with your other functions!)

File: ives_libby_ndvi.py [hunk 4]
```diff
@@ -0,0 +1,1615 @@
...
+    "    temp_df['date'] = pd.to_datetime(temp_df['date'])\n",
+    "    temp_df = temp_df.set_index('date')\n",
+    "\n",
+    "    return temp_df\n",
+    "\n",
+    "\n",
+    "\n",
+    "def combine_tifs(tif_list):\n",
+    "    \"\"\"A function that combines a list of tifs in the same CRS\n",
+    "    and of the same extent into an xarray object\n",
```
Comment (gnarledbranches) [inline ives_libby_ndvi.py:541]: Great! Clear and succinct summary of this function.

File: ives_libby_ndvi.py [hunk 5]
```diff
@@ -0,0 +1,1615 @@
...
+    "        out_xr.append(rxr.open_rasterio(tif_path, masked=True).squeeze())\n",
+    "        out_xr[i][\"band\"] = i+1\n",
+    "\n",
+    "    return xr.concat(out_xr, dim=\"band\")\n",
+    "\n",
+    "\n",
+    "def mask_clouds(input_path):\n",
+    "    \"\"\"\n",
+    "    This function uses the landsat qa.tif to mask areas on the image that are\n",
+    "    likely clouds, cloud shadows, or shadows.\n",
```
Comment (gnarledbranches) [inline ives_libby_ndvi.py:566]: Great! Clear and succinct summary of this function.

File: ives_libby_ndvi.py [hunk 6]
```diff
@@ -0,0 +1,1615 @@
...
+    "fig, ax = plt.subplots(figsize=(15, 10))\n",
+    "ax.set_title(\n",
+    "    'Mean Normalized Different Vegetation Index (NDVI)\\nJan 2017 - Dec 2017\\nLandsat8 with Clouds Removed')\n",
+    "\n",
+    "for site in sites:\n",
+    "    temp_df = results[results[\"site\"] == site]\n",
+    "    ax.plot(temp_df.index.values,\n",
+    "            temp_df['mean_ndvi'],\n",
+    "            'o-',\n",
+    "            label=site)\n",
```
Comment (gnarledbranches) [inline ives_libby_ndvi.py:1297]: Nice! My own plot code could be improved by doing this.

**Review by gnarledbranches — COMMENTED** [review]
@lrwives

**Comment by lrwives** [comment]
@gnarledbranches Thank you for the review! There are definitely a few parts of the assignment that I missed in the interest of time. Glad you liked the looping for plots. I did have trouble not plotting NaNs, but doubt that was a problem with the loop...

**Pull request closed without merging**